sdk/Pulumi/Provider/Provider.cs

```diff
@@ -540,6 +540,21 @@ private ImmutableDictionary<string, PropertyValue> Unmarshal(Struct? properties)
             return PropertyValue.Unmarshal(properties);
         }
 
+        // Helper to marshal CheckFailures from the domain to the GRPC layer.
+        private IEnumerable<Pulumirpc.CheckFailure> mapFailures(IEnumerable<CheckFailure>? failures)
```

Comment:
Can you rename this to `MapFailures` instead? (as per C# standards)

Reply:
I switch languages too much 😆